### PR TITLE
docs/agent: merge template and template config

### PR DESCRIPTION
**Pull request by jasonodonnell**
This merges the `template_config` and `template` stanza documentation into a single page, broken into global and per template sections.

### CODE REVIEW

File: website/content/docs/agent/template.mdx [hunk 1]
```diff
@@ -80,50 +142,13 @@ The top level `template` block has multiple configurations entries:
...
-```
-{{ with secret "secret/my-secret" }}
-{{ .Data.data.foo }}
-{{ end }}
-```
 
-## Example Configuration
+### Example
 
 The following demonstrates configuring Vault Agent to template secrets using the
```
Comment (swenson) [inline website/content/docs/agent/template.mdx:148]: I think this paragraph no longer applies to this example.

File: website/content/docs/agent/template.mdx [hunk 2]
```diff
@@ -171,3 +196,45 @@ using the certificates `validTo` field.
 
 This does not apply to certificates generated with `generate_lease: true`. If set
 Vault Agent template will apply the non-renewable, leased secret rules.
+
+## Example
```
Comment (swenson) [inline website/content/docs/agent/template.mdx:200]: Should this header be updated to be more specific than "Example"?

File: website/content/docs/agent/template.mdx (hunk 1 above)
Comment (jasonodonnell) [reply website/content/docs/agent/template.mdx]: Good catch!

File: website/content/docs/agent/template.mdx (hunk 2 above)
Comment (jasonodonnell) [reply website/content/docs/agent/template.mdx]: Updated.

**Review by swenson — APPROVED** [review]
LGTM. It's much better to have these in one place. :)

**Pull request merged**